### PR TITLE
Removed use of tempfile persist.

### PR DESCRIPTION
**Pull request by thewizzy**
For users who have their project on a different storage device to their temp directory it causes a panic.

The code creates a temporary file, then persists it to the data directory only to remove it within the test. This seems superfluous.

The current setup just uses the tempfile where it stands and relies on the destructors in tempfile to remove files when they fall out of scope. If there is a better way to convert the `tmpfile.path` to a `String`, I would appreciate the input.

I have tested the Solution tests on OSX, Windows and Linux. All passed.

Example error:
```
--- db::tests::database::read_db_should_fail_with_invalid_json stdout ----
thread 'db::tests::database::read_db_should_fail_with_invalid_json' panicked at 'called `Result::unwrap()` on an `Err` value: PathPersistError { error: Os { code: 17, kind: CrossesDevices, message: "The system cannot move the file to a different disk drive." }, path: "C:\\Users\\...\\AppData\\Local\\Temp\\.tmpvnwYSi" }', src\db.rs:56:38
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### CODE REVIEW

**Comment by thewizzy** [comment]
Any feedback?
I understand this is a corner case, but at least 2 people have hit it. Basically only happens when your code is on  a different device to your tempdir.

File: 4. Projects/1. CLI/Problem/Stage 1/Step 2/src/db.rs [hunk 1]
```diff
@@ -108,10 +91,7 @@ mod tests {
             let write_result = db.write_db(&state);
             let read_result = db.read_db().unwrap();
 
-            remove_file(file_path).unwrap();
-
             assert_eq!(write_result.is_ok(), true);
-            // TODO: fix this error by deriving the appropriate traits for DBState
```
Comment (letsgetrusty) [inline 4. Projects/1. CLI/Problem/Stage 1/Step 2/src/db.rs:114]: Why did you remove this comment?

File: 4. Projects/1. CLI/Problem/Stage 2/Step 2/src/db.rs [hunk 2]
```diff
@@ -457,10 +440,7 @@ mod tests {
             let write_result = db.write_db(&state);
             let read_result = db.read_db().unwrap();
 
-            remove_file(file_path).unwrap();
-
             assert_eq!(write_result.is_ok(), true);
             assert_eq!(read_result, state);
-        }
-    }
+        }    }
```
Comment (letsgetrusty) [inline 4. Projects/1. CLI/Problem/Stage 2/Step 2/src/db.rs:445]: fix formatting

**Review by letsgetrusty — CHANGES_REQUESTED** [review]
Overall LGTM. Thank you for the contribution. Please address my comments and I'll get this merged.

File: 4. Projects/1. CLI/Problem/Stage 1/Step 2/src/db.rs (hunk 1 above)
Comment (thewizzy) [reply 4. Projects/1. CLI/Problem/Stage 1/Step 2/src/db.rs:114]: Sorry, there were a lot of files that shared the same tests, I copy and pasted the relevant tests and didn't realize this was in that stage!

**Review by letsgetrusty — APPROVED** [review]
LGTM!
